src/repl/scala/tools/nsc/interpreter/package.scala

```diff
@@ -88,8 +88,12 @@ package object interpreter extends ReplConfig with ReplStrings {
           }
       }
 
-      if (filtered.isEmpty)
+      if (filtered.isEmpty) {
+        if (global.settings.nopredef || global.settings.noimports) {
+          return "No implicits have been imported."
+        }
         return "No implicits have been imported other than those in Predef."
+      }
```

Comment:
Is it desirable to get rid of the return statements? 
Right now you know the 'termination' of the method if theseconditions are met.

(I know the foreach won't do anything if the conditions are met as well, but could make it harder to read through the end of the method).